Description: Remove winetricks built-in update functionality.
Author: Joseph Bisch <joseph.bisch@gmail.com>
Author: Jens Reyer <jre.winesim@gmail.com>
Forwarded: not-needed
Last-Update: 2017-04-11
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/winetricks
+++ b/src/winetricks
@@ -929,6 +929,9 @@ w_dotnet_verify()
 # Checks if the user can run the self-update/rollback commands
 winetricks_check_update_availability()
 {
+    w_warn "Self-update not available. Use your distribution's package management."
+    exit
+
     # Prevents the development file overwrite:
     if test -d "../.git"; then
         w_warn "You're running in a dev environment. Please make a copy of the file before running this command."
@@ -3142,6 +3145,9 @@ winetricks_get_platform()
 
 winetricks_latest_version_check()
 {
+    w_info "winetricks latest version check update disabled by your distribution"
+    return
+
     if [ "$WINETRICKS_LATEST_VERSION_CHECK" = 'disabled' ] || [ -f "${WINETRICKS_CONFIG}/disable-latest-version-check" ] ; then
         w_info "winetricks latest version check update disabled"
         return
@@ -5230,7 +5236,6 @@ winetricks_init()
         echo "Using winetricks $(winetricks_print_version) with ${WINETRICKS_ORIG_WINE_VERSION} and WINEARCH=${W_ARCH}"
     fi
 
-    winetricks_latest_version_check
 }
 
 winetricks_usage()
@@ -5304,8 +5309,6 @@ Options:
     --force           Don't check whether packages were already installed
     --gui             Show gui diagnostics even when driven by commandline
     --isolate         Install each app or game in its own bottle (WINEPREFIX)
-    --self-update     Update this application to the last version
-    --update-rollback Rollback the last self update
 -k, --keep_isos       Cache isos (allows later installation without disc)
     --no-clean        Don't delete temp directories (useful during debugging)
 -q, --unattended      Don't ask any questions, just install automatically
@@ -5351,10 +5354,8 @@ winetricks_handle_option()
         --optout) WINETRICKS_STATS_REPORT=0;;
         -q|--unattended) winetricks_set_unattended 1 ;;
         -r|--ddrescue) WINETRICKS_OPT_DD=ddrescue ;;
-        --self-update) winetricks_selfupdate;;
         --showbroken) W_OPT_SHOWBROKEN=1 ;;
         -t|--torify)  WINETRICKS_OPT_TORIFY=1 ;;
-        --update-rollback) winetricks_selfupdate_rollback;;
         -v|--verbose) WINETRICKS_OPT_VERBOSE=1 ; set -x;;
         -V|--version) winetricks_print_version ; exit 0;;
         --verify) WINETRICKS_VERIFY=1 ;;
